### PR TITLE
Feat/handle loading

### PR DESCRIPTION
**Pull request by tranchinh123**

### CODE REVIEW

File: bookshelter-app/src/components/BookCard/BookCard.stories.tsx [hunk 1]
```diff
@@ -2,18 +2,15 @@
...
   component: BookCard,
 } as Meta;
 
 interface BookCardArgs {
-  id: string;
-  title: string;
-  author: string;
-  published: number;
-  cover: string;
+  Book: Book;
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/BookCard/BookCard.stories.tsx]: `book`

Comment (huypham-agilityio) [reply bookshelter-app/src/components/BookCard/BookCard.stories.tsx]: similar check for all

File: bookshelter-app/src/components/BookCategory/index.tsx [hunk 2]
```diff
@@ -1,26 +1,19 @@
 import styles from './index.module.css';
-interface CateGoryItemProps {
-  category: string;
-  quantity: number;
+import { Category } from '../../types';
+interface BookCategoryProps {
+  Category: Category;
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/BookCategory/index.tsx]: same here

File: bookshelter-app/src/components/BookCard/BookCard.stories.tsx (hunk 1 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/BookCard/BookCard.stories.tsx]: fixed!

File: bookshelter-app/src/components/BookCategory/index.tsx (hunk 2 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/BookCategory/index.tsx]: Fixed!

File: bookshelter-app/src/pages/Detail/index.tsx [hunk 3]
```diff
@@ -15,22 +14,26 @@ const DetailPage = () => {
 
   useEffect(() => {
     const fetchBookList = async (): Promise<void> => {
```
Comment (huypham-agilityio) [inline bookshelter-app/src/pages/Detail/index.tsx]: Update naming, this is only fetchBookDetail

File: bookshelter-app/src/pages/Home/index.tsx [hunk 4]
```diff
@@ -41,30 +42,14 @@ const HomePage = () => {
...
-  return (
-    <DefaultLayout
-      isFilteredSlug={isFilteredSlug}
-      isFilteredName={isFilteredName}
-      books={books}
-    >
-      {<BookCardList books={books} />}
-    </DefaultLayout>
-  );
+  return <>{loading ? <Loading /> : <BookCardList books={books} />}</>;
```
Comment (huypham-agilityio) [inline bookshelter-app/src/pages/Home/index.tsx]: why need <>

File: bookshelter-app/src/pages/NotFound/index.tsx [hunk 5]
```diff
@@ -1,14 +1,13 @@
-import DefaultLayout from '../../layouts';
 import Glass from '../../components/Icons/Glass';
 import styles from './index.module.css';
 const NotFound = () => {
   return (
-    <DefaultLayout isFilteredSlug={false} isFilteredName={false} books={[]}>
+    <>
```
Comment (huypham-agilityio) [inline bookshelter-app/src/pages/NotFound/index.tsx]: same here

Comment (tranchinh123) [reply bookshelter-app/src/pages/NotFound/index.tsx]: fixed!

File: bookshelter-app/src/pages/Detail/index.tsx (hunk 3 above)
Comment (tranchinh123) [reply bookshelter-app/src/pages/Detail/index.tsx]: Fixed!

File: bookshelter-app/src/pages/Home/index.tsx (hunk 4 above)
Comment (tranchinh123) [reply bookshelter-app/src/pages/Home/index.tsx]: fixed!

**Pull request merged**